### PR TITLE
ADEN-12189 Integrate LiveConnect into f2

### PR DESCRIPTION
**Pull request by szymon-nawrocki**
### Ticket
https://fandom.atlassian.net/browse/ADEN-12189

### CODE REVIEW

**Review by domjanik — APPROVED** [review]

**Comment by domjanik** [comment]
Looks good to me :)

File: src/ad-services/live-connect/index.ts [hunk 1]
```diff
@@ -39,6 +43,8 @@ class LiveConnect extends BaseServiceSetup {
 			return;
 		}
 
+		this.isLoaded = this.isAvailableInLocalStorage(`${partnerName}-unifiedId`);
```
Comment (mfarul) [inline src/ad-services/live-connect/index.ts:46]: Here we should check for unifiedId and sha2 presence as the first one will be almost always available but sha2 could be resolved later. 
Then I would also add checking optional once in resolveId.

**Review by mfarul — CHANGES_REQUESTED** [review]

**Review by mfarul — APPROVED** [review]

File: src/ad-services/live-connect/index.ts (hunk 1 above)
Comment (mfarul) [reply src/ad-services/live-connect/index.ts:46]: On the other hand we use session storage here, so the chance of resolving sha2 during session is almost 0%. Let's leave it as it is.

**Review by domjanik — APPROVED** [review]

File: src/ad-services/live-connect/index.ts [hunk 2]
```diff
@@ -49,23 +56,27 @@ class LiveConnect extends BaseServiceSetup {
...
 					this.track();
 				});
-
-			this.isLoaded = true;
+		} else {
+			utils.logger(logGroup, `already loaded and available in ${this.storageConfig.type}Storage`);
 		}
 	}
 
 	resolveId(idName, partnerName) {
```
Comment (mfarul) [inline src/ad-services/live-connect/index.ts:64]: shouldn't we skip resolving if we have ID in cache?

File: src/ad-services/live-connect/index.ts [hunk 3]
```diff
@@ -92,6 +103,67 @@ class LiveConnect extends BaseServiceSetup {
...
+			return this.storage.get(key);
+		} else {
+			return this.storage.getItem(key);
+		}
+	}
+
+	saveToStorage(key: string, value: string) {
+		const idName = this.getConfigId(key);
+
+		if (!this.storageConfig.mandatoryParams.includes(idName)) {
```
Comment (mfarul) [inline src/ad-services/live-connect/index.ts:149]: why saving only mandatory? those are only set up to skip or fire resolving

**Pull request merged**